### ISSUE TITLE
[2020 NeurIPS] Learning from Failure: Training Debiased Classifier from Biased Classifier

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#56]
해당 논문은 2가지 main contribution

1. Neural network는 target attribute를 배우는 것보다 bias를 배우는 것이 쉬운 경우, bias를 학습하며 이는 학습 초기에 두드러지게 나타남
2. 논문에서 제시한 LfF (Learning from Failure)를 통해 bias를 labeling, predefine하지 않아도 debiased representation을 배울 수 있음

### 1.1. Dataset
<img width="1104" alt="image" src="https://user-images.githubusercontent.com/49643709/159265928-480a8045-2441-45d5-89c3-b0d3aab93e60.png">

2 Attributes of each dataset
- Colored MNIST : digit, color
- Corrupted CIFAR-10 : category of the object, type of corruption

### 1.2. Not all biases are malignant
<img width="1144" alt="image" src="https://user-images.githubusercontent.com/49643709/159266948-cd033e43-bb2c-49dc-982a-c0177e0b40f2.png">

모델을 biased dataset에 학습시킨다고해서 반드시 bias를 배우는 것이 아님

Target attribute를 배우는 것보다 bias를 배우는 것이 더 쉽다면 모델이 bias를 배우게 되며, 이를 malignant bias라고 함
(반대의 경우 benign bias)

### 1.3. Malignant bias is learned first
<img width="1114" alt="image" src="https://user-images.githubusercontent.com/49643709/159267833-c216c402-1c1f-4deb-9da7-311b0299843f.png">

Malignant bias인 경우, bias-aligned samples와 bias-conflicting samples의 loss dynamic이 매우 다름
- Bias-aligned samples loss는 빠르게 0으로 감소함
- 그에 반해 bias-conflicting samples loss는 증가하다가 bias-aligned samples loss가 0에 가까워져야 감소하기 시작함

Neural network가 easy concept인 malignant bias를 먼저 배운다는 점에서 noisy dataset에서 학습하는 것과 비슷함
(neural network는 easy concept를 먼저 학습하고, 그 다음에 hard concept를 배움)

## 2. Debiasing by learning from failure (LfF)
<img width="1144" alt="image" src="https://user-images.githubusercontent.com/49643709/159270196-7e070174-9ad8-4be7-80e8-bb6af0946e7d.png">
<img width="909" alt="image" src="https://user-images.githubusercontent.com/49643709/159270227-9c7649f5-ae2e-471e-b0e8-c2225d066994.png">

Key concept는 다음과 같음

Pair of neural networks (f_B, f_D)를 학습하는데 f_B는 의도적으로 biased되게 학습하며, f_D는 biased model이 잘 배우지 못하는 sample에 집중해서 훈련

### 2.1. Training a biased model
<img width="1115" alt="image" src="https://user-images.githubusercontent.com/49643709/159273371-61608501-997c-4268-9fa4-52efcbcd597a.png">

GCE (generalized cross entropy) loss는 원래 noisy dataset에서 학습에 사용되던 loss임

식을 보면 label에 해당하는 model의 probability 값이 클수록 gradient 값이 커지며, 이는 noisy하지 않은 correct sample을 학습하도록 만듬

Biased model f_B를 GCE loss로 학습하여 의도적으로 biased되게 만듬

참고로 논문에서는 GCE loss 논문과 같이 q를 0.7로 설정하여 모든 실험 진행

### 2.2. Training a debiased model
<img width="534" alt="image" src="https://user-images.githubusercontent.com/49643709/159275357-3b374385-f903-4809-973e-911cef1904ac.png">

Relative difficulty score는 sample이 bias-conflicting sample인 가능성이 어느 정도인지를 측정하며, bias-conflicting sample일 경우 값이 크고 아니면 작아짐

Relative difficulty score를 이용하여 debiased model f_D를 학습

결국 작동 원리는 중요한 sample에 대하여 가중치를 많이 줘서 학습하는 것임
(bias를 정의하지 않았으며 라벨이 없기에, 어떤 sample이 bias-conflict인지 모르기 때문에 oversampling같은 정보를 사용할 수 없음)

#### 3.1.1. Baselines
<img width="1118" alt="image" src="https://user-images.githubusercontent.com/49643709/159282040-bac45f6e-8262-4f1f-aa70-3704eafa548a.png">
<img width="1137" alt="image" src="https://user-images.githubusercontent.com/49643709/159282087-95220611-82c2-4f63-97d0-1b7f36091dc2.png">
<img width="1120" alt="image" src="https://user-images.githubusercontent.com/49643709/159282182-b118d3ef-0cc2-455c-8e81-beee0049620a.png">
<img width="1127" alt="image" src="https://user-images.githubusercontent.com/49643709/159282217-e7438b8d-4212-4164-9176-fb23b3f8d87e.png">

Our algorithm LfF consistently outperform the baseline algorithm by a large margin, regardless of the difficulty for the biased attribute or ratio for bias-aligned samples

#### 3.1.2. Detailed analysis of failure-based debiasing
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/49643709/159283976-a2241ebf-d35a-496f-af97-423201e5d64f.png">

Biased model은 bias-aligned samples에서 accuracy가 거의 100%인데 bias-conflicting samples에서 accuracy는 random guessing에 가까움

즉 biased model은 의도한대로 bias attribute를 잘 학습했음

반대로 debiased model은 bias-aligned / bias-conflicting samples간의 accuracy gap이 거의 없고 성능도 좋음

이로부터 debiased model이 bias attribute를 학습하지 않고 intended target attribute를 잘 학습했다는 것을 알 수 있음

그에 반해 vanilla model은 bias-aligned / bias-conflicting samples간의 accuracy gap이 크고 bias-conflicting samples에서 성능이 낮음

이로부터 standard cross entropy loss로 학습한 vanilla model은 bias attribute과 더불어 target attribute의 일부를 학습한다고 볼 수 있음

#### 3.1.3. Contribution of GCE loss
<img width="1127" alt="image" src="https://user-images.githubusercontent.com/49643709/159284018-102b61da-6088-4c03-ac20-ed7f32d0ff7f.png">

GCE loss로 bias model을 학습시키는게 아니라 standard CE loss로 bias model을 학습

위에서 언급했지만 standard CE loss로 학습하면 bias attribute만 학습하는 것이 아니라 target attribute의 일부를 학습하기에 debiasing model의 성능을 낮출 것이라 예측할 수 있음

실험 결과도 이를 증명하며, GCE loss로 biased model을 학습시키는 것이 중요함을 알 수 있음

#### 3.1.4. Comparison to other combination rule
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/49643709/159282390-47315915-0fe3-4c19-bc3b-eaa06c32b6c6.png">

Logits나 prediction을 직접 조절하는 방법보다는 resampling / reweighting 방법이 더 효과적임

#### 3.2.1. CelebA
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/49643709/159290628-e7803d19-c3d0-438c-aca1-e48f0f4df3cb.png">

CelebA dataset에서 gender와 hair color, heavy makeup attribute 간의 high correlation을 발견함

Hair color, heavy makeup을 target, gender를 bias로 설정하여 dataset 구축

Our algorithm LfF consistently outperforms the vanilla model while the vanilla model suffers from gender bias existing in the real-world dataset

Our model is comparable to Group DRO, which requires explicit labeling for the bias attribute (unlike ours)

#### 3.2.2. Biased action recognition dataset
<img width="1137" alt="image" src="https://user-images.githubusercontent.com/49643709/159291076-a598b4ce-ffee-42b1-a952-a66fd07b0ba9.png">
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/49643709/159291141-e18192f5-d98e-449f-a34c-0ab22989dd70.png">

Place-biased action recognition (BAR) dataset을 구축하여 실험

Our algorithm LfF outperforms the vanilla classifier for all action classes

LfF outperforms ReBias, which is also free from explicit labeling on the bias attribute, for most action classes

**Comment by Jasonlee1995** [comment]
<img width="1323" alt="image" src="https://user-images.githubusercontent.com/49643709/159294427-6f6fde74-451b-4361-9de9-a434eeeddf96.png">

Stable train을 위해 loss의 exponential moving average를 사용하여 relative difficulty score 계산
